src/app/drive/components/FileViewer/FileViewer.tsx

```diff
@@ -162,17 +174,23 @@ const FileViewer = ({
   useEffect(() => {
     setIsPreviewAvailable(true);
     const largeFile = isLargeFile(file?.size);
-
+    trackFilePreviewClicked(trackFilePreviewProperties);
```

Comment:
I think that this place for these events is not the best, it is triggered every time the file is displayed or changed and not when the button is clicked to show the file preview.

Reply:
> I think that this place for these events is not the best, it is triggered every time the file is displayed or changed and not when the button is clicked to show the file preview.

I have been testing it and it only executes when it corresponds to the event requirements.

Reply:
This event is created to detect possible extensions that do not have preview and in case it is necessary to implement it. I understand that it must be executed whenever a file is previewed regardless of how it is accessed.